Makefile

```diff
@@ -1,9 +1,359 @@
+L3_CC_DATA          := $(TMPDIR)/$(L3PACKAGE).cc-$(TEST_DATA_SUFFIX)
+L3_CC_SMALL_DATA    := $(TMPDIR)/$(L3PACKAGE).cc-small-$(TEST_DATA_SUFFIX)
+
+# ###################################################################
+# ---- Symbols to build test-code sample programs
+# ###################################################################
+
+# -- Pick-up both C and C++ sources in separate lists, named by a symbol.
+TEST_C_CODE_SRC := $(shell find $(L3_SRCDIR) $(TESTSDIR) -type f \( -name *.c \) -print)
+
+# Need to pick-up l3.c also. (Support both *.cpp and *.cc for C++).
+TEST_CPP_CODE_SRC := $(shell find $(L3_SRCDIR) $(TESTSDIR) -type f \( -name l3.c -o -name *.cpp -o -name *.cc \) -print)
+TEST_CC_CODE_SRC  := $(shell find $(L3_SRCDIR) $(TESTSDIR) -type f \( -name l3.c -o -name *.cc -o -name *.cc \) -print)
+
+# Uncomment this to see symbol, for debugging.
+# $(info $$TEST_C_CODE_SRC     = [${TEST_C_CODE_SRC} ])
+# $(info $$TEST_CPP_CODE_SRC   = [${TEST_CPP_CODE_SRC} ])
+
+# ----------------------------------------------------------------------------
+# Generate location / name of test-use-cases sample program binary, using
+# built-in substitution references. test-use-cases/ has sources like:
+#
+#   test-use-cases/single-file-C-program/test-main.c
+#   test-use-cases/single-file-Cpp-program/test-main.cpp
+#
+# Convert this list of *main.c to generate test-code program binary names,
+# using the src-dir's name as the program name.
+#
+# Generate, respectively:
+#
+#   build/release/bin/test-use-cases/single-file-C-program
+#   build/release/bin/test-use-cases/single-file-Cpp-program
+#
+TEST_C_CODE_BINS_TMP := $(TEST_C_CODE_SRC:$(TESTSDIR)/%-main.c=$(BINDIR)/$(TESTSDIR)/%-program)
+TEST_C_CODE_BINS=$(patsubst %program/, %program, $(dir $(TEST_C_CODE_BINS_TMP)) )
+
+# Uncomment this to see symbol, for debugging.
+# $(info $$TEST_C_CODE_BINS     = [${TEST_C_CODE_BINS} ])
+
+TEST_CPP_CODE_BINS_TMP := $(TEST_CPP_CODE_SRC:$(TESTSDIR)/%-main.cpp=$(BINDIR)/$(TESTSDIR)/%-program)
+TEST_CPP_CODE_BINS_TMP := $(TEST_CPP_CODE_BINS_TMP:$(TESTSDIR)/%-main.cc=$(BINDIR)/$(TESTSDIR)/%-program)
+TEST_CPP_CODE_BINS=$(patsubst %program/, %program, $(dir $(TEST_CPP_CODE_BINS_TMP)) )
+
+TEST_CC_CODE_BINS_TMP := $(TEST_CPP_CODE_SRC:$(TESTSDIR)/%-main.cpp=$(BINDIR)/$(TESTSDIR)/%-program)
+TEST_CC_CODE_BINS_TMP := $(TEST_CPP_CODE_BINS_TMP:$(TESTSDIR)/%-main.cc=$(BINDIR)/$(TESTSDIR)/%-program)
+TEST_CC_CODE_BINS=$(patsubst %program/, %program, $(dir $(TEST_CPP_CODE_BINS_TMP)) )
+
+# Uncomment this to see symbol, for debugging.
+# $(info $$TEST_CPP_CODE_BINS     = [${TEST_CPP_CODE_BINS} ])
+
+# Build symbol for single C & C++ unit-test binary that we run
+C_UNIT_TEST_BIN     := $(BUILD_ROOT)/$(BUILD_MODE)/bin/test-use-cases/single-file-C-program
+CPP_UNIT_TEST_BIN   := $(BUILD_ROOT)/$(BUILD_MODE)/bin/test-use-cases/single-file-Cpp-program
+CC_UNIT_TEST_BIN    := $(BUILD_ROOT)/$(BUILD_MODE)/bin/test-use-cases/single-file-CC-program
+
```

Comment:
Each of the 3 generated test program binaries are exercised as part of the test targets.

**NOTE: Heads-up**: For next steps of integration with LOC, I plan to add test programs with 2-files, and possibly with multiple-files. This provides the sample-test scaffolding to better demonstrate how you can unpack LOC value into constituent different file names and line numbers.

Hence the choice of names like `single-file-CC-program` and so on.